File: test-suite-python/src/test/python/micronaut/security/docs/securityrule/permitall/ExampleController.py
# tag::exampleControllerPlusImports[]
from jakarta.annotation.security import PermitAll, RolesAllowed
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get, Produces
# end::exampleControllerPlusImports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="PermitAllTest")
# tag::exampleControllerPlusImports[]

@Controller("/example")
class ExampleController:

    @Produces(MediaType.TEXT_PLAIN)
    @Get("/admin")
    @RolesAllowed(["ROLE_ADMIN", "ROLE_X"])  # <1>
    def withroles(self) -> str:
        return "You have ROLE_ADMIN or ROLE_X roles"

    @Produces(MediaType.TEXT_PLAIN)
    @Get("/anonymous")
    @PermitAll  # <2>
    def anonymous(self) -> str:
        return "You are anonymous"
# end::exampleControllerPlusImports[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/securityrule/permitall/ExampleController.kt
package io.micronaut.security.docs.securityrule.permitall

//tag::exampleControllerPlusImports[]
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import jakarta.annotation.security.PermitAll
import jakarta.annotation.security.RolesAllowed
//end::exampleControllerPlusImports[]
import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "PermitAllTest")
//tag::exampleControllerPlusImports[]

@Controller("/example")
class ExampleController {

    @Produces(MediaType.TEXT_PLAIN)
    @Get("/admin")
    @RolesAllowed("ROLE_ADMIN", "ROLE_X") // <1>
    fun withroles(): String = "You have ROLE_ADMIN or ROLE_X roles"

    @Produces(MediaType.TEXT_PLAIN)
    @Get("/anonymous")
    @PermitAll  // <2>
    fun anonymous(): String = "You are anonymous"
}
//end::exampleControllerPlusImports[]
